### PR TITLE
feat: download button on 'Pricing' page 🔼

### PR DESCRIPTION
**Pull request by wflore19**
### Description

- Adds functional download button

### Related Issue

[#164 ](https://github.com/trypear/pear-landing-page/issues/164)

### Changes Made

- Created an api route to request download url from server
- added button to trigger request
- shows the button that relates to the users OS

### Screenshots

image

### Checklist

- [x] I have tagged the issue in this PR.
- [x] I have attached necessary screenshots.
- [x] I have provided a short description of the PR.
- [x] I ran `yarn build` and build is successful
- [x] My code follows the style guidelines of this project.
- [ ] I have added necessary documentation (if applicable)

### CODE REVIEW

**Comment by nang-dev** [comment]
Yo! ill be taking a look at this!

File: components/pricing.tsx [hunk 1]
```diff
@@ -45,28 +123,38 @@ const PricingTier: React.FC<PricingTierProps> = ({
...
+            {!user && (
+              <p>
+                Please{" "}
+                <a href="/signin" className="text-link">
+                  log in
+                </a>{" "}
+                to download for free.
+              </p>
+            )}
+          </>
```
Comment (Himanshu-Singh-Chauhan) [inline components/pricing.tsx:146]: should we login even to download for free?

File: components/pricing.tsx [hunk 2]
```diff
@@ -45,28 +123,38 @@ const PricingTier: React.FC<PricingTierProps> = ({
...
-          ))
+          numberOfDownloads < 3 ? (
+            <>
+              <DownloadButton os={getOS()} />
+            </>
+          ) : (
+            <p className="text-sm font-medium text-gray-400 sm:text-base">
+              Thank you for downloading PearAI!
+            </p>
+          )
```
Comment (Himanshu-Singh-Chauhan) [inline components/pricing.tsx:157]: if numberOfDownloads is greater than 3, and user comes few days later to download the update, does he see download button or not? 
right now we don't push updates directly, so user have to download it from here again.

**Comment by wflore19** [comment]
@Himanshu-Singh-Chauhan I think we need users to login for download, @Fryingpannn ? I removed the download condition logic. 

When downloaded the button will be replaced with "Thanks for downloading". Refresh will show the download button again

File: components/pricing.tsx (hunk 1 above)
Comment (nang-dev) [reply components/pricing.tsx:125]: ^ Yes correct, we shouldn't

Comment (Fryingpannn) [reply components/pricing.tsx:152]: Yes for now! (due to waitlist)

**Comment by Fryingpannn** [comment]
Pushed commits, Updated

**Review by Fryingpannn — APPROVED** [review]
Just missing spinner! (loading effect when checking if user is on waitlist)

**Comment by wflore19** [comment]
Spinner added

**Review by Fryingpannn — APPROVED** [review]

**Pull request merged**